src/kernel_2.jl

```diff
@@ -1,27 +1,40 @@
 function key_sampling(
-  g::SimpleWeightedGraph{T,R}
-  ) where T<:Integer where R<:Real
+  g::Graph{T}
+  ) where T<:Integer
 
   n_v = nv(g)
-  rand_keys = sample(1:n_v, 64, replace = false)
+  rand_keys = randperm(n_v)
```

Comment:
This is really expensive. Consider using the `sample!([rng, ]a, k)` code in LightGraphs, in `src/utils.jl`. Since you need 64 vertices with degree > 0, sample 128, then test, then keep sampling by the number you still need * 2 until you get 64.